internal/resource/resource.go

```diff
@@ -0,0 +1,131 @@
+package resource
```

Comment:
This file deserves another pass of refactoring and some unit tests - will address in a separate PR.